src/App.tsx

```diff
@@ -1,27 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [visibleGoods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleClick = (getGoods: () => Promise<Good[]>) => {
+    getGoods().then((goods) => setGoods(goods));
```

Comment:
since `setGoods()` expects the same argument as the `then()` method callback function, it can be this simpler
```suggestion
    getGoods().then(setGoods);
```

Reply:
also, don't forget to handle errors on FE, show them, etc.